packages/functions/src/dal/AppointmentDataAccessLayer.ts

```diff
@@ -161,3 +161,13 @@ export function removeDonorFromDbAppointment(
     lastChangeType: BookingChange.CANCELLED,
   };
 }
+
+export function confirmArrivedFromDbAppointment(
+  appointment: DbAppointment
+): DbAppointment {
+  const { ...otherProperties } = appointment;
+  return {
+    ...otherProperties,
+    donationDoneTimeMillis: admin.firestore.Timestamp.now(),
+  };
```

Comment:
I think it would be simpler to just do 
`appointment.donationTimeMillis = admin.firestore.Timestamp.now()`

There's no need for this object distruction.

What do you think?